talaiot/src/main/kotlin/com/cdsap/talaiot/metrics/GradleMetrics.kt

```diff
@@ -1,63 +1,122 @@
+        daemonScanInfo?.isSingleUse?.toString()
+    },
+    assigner = { report, value -> report.environment.switches.daemon = value }
+)
+
+class GradleMaxWorkersMetric : GradleMetric<String>(
+    provider = { it.gradle.startParameter.maxWorkerCount.toString() },
+    assigner = { report, value -> report.environment.maxWorkers = value }
+)
+
+class JvmXmxMetric() : JvmArgsMetric(
+    argProvider = { paramList: List<String> ->
+        val xmxParam = paramList.find { param -> param.contains("Xmx") }
+        xmxParam?.split("Xmx")?.get(1)?.toBytes()
+    },
+    assigner = { report, value -> report.environment.javaXmxBytes = value }
+)
+
+class JvmXmsMetric() : JvmArgsMetric(
+    argProvider = { paramList: List<String> ->
+        val xmsParam = paramList.find { param -> param.contains("Xms") }
+        xmsParam?.split("Xms")?.get(1)?.toBytes()
+    },
+    assigner = { report, value -> report.environment.javaXmsBytes = value }
+)
+
+class JvmMaxPermSizeMetric() : JvmArgsMetric(
+    argProvider = { paramList: List<String> ->
+        val maxPermSize = paramList.find { param -> param.contains("MaxPermSize") }
+        maxPermSize?.split("=")?.get(1)?.toBytes()
+    },
+    assigner = { report, value -> report.environment.javaMaxPermSize = value }
+)
+
+class GradleBuildCacheModeMetric : GradleMetric<String>(
+    provider = {
+        val settings = (it.rootProject as ProjectInternal).gradle.settings
+        when {
+            settings.buildCache.remote == null -> "local"
+            else -> "remote"
         }
-        gradleMetrics["gradleVersion"] = project.gradle.gradleVersion
-        return gradleMetrics
-
-    }
-
-    /**
-     * Check if a property exists in the Gradle project given a property name
-     * @param property name of the property to check
-     * @return if the property is present in the configuration
-     */
-    private fun hasProperty(property: String) = project.gradle.rootProject.hasProperty(property)
-
-
-    /**
-     * Get property from the Gradle project given a property name
-     * @param property name of the property to retrieve
-     * @return value of the property
-     */
-    private fun property(property: String) = project.property(property)
-}
\ No newline at end of file
+    },
+    assigner = { report, value -> report.environment.cacheMode = value }
+)
+
+class GradleBuildCachePushEnabled : GradleMetric<String?>(
+    provider = {
+        val settings = (it.rootProject as ProjectInternal).gradle.settings
+        settings.buildCache.remote?.isPush?.toString()
+    },
+    assigner = { report, value -> report.environment.cachePushEnabled = value }
+)
+
+class GradleRequestedTasksMetric : GradleMetric<String>(
+    provider = {
+        it.gradle.startParameter.taskNames.joinToString(separator = " ")
+    },
+    assigner = { report, value -> report.requestedTasks = value}
+)
+
+//class GradleScanUrlMetric : GradleMetric<String>(
```

Comment:
can we remove these commented code?

Reply:
Actually moved part of the metric generation from Listener. So, fixed